pytorch_lightning/metrics/functional/classification.py

```diff
@@ -104,9 +107,31 @@ def stat_scores_multiple_classes(
         >>> sups
         tensor([1., 0., 1., 1.])
 
+        By given pos_labels [1, 2], we will get x = [1, 1, 0] and y = [0, 1, 0]
+        >>> x = torch.tensor([0, 1, 2, 3])
+        >>> y = torch.tensor([0, 1, 2, 2])
+        >>> pos_labels = torch.tensor([2])
+        >>> tps, fps, tns, fns, sups = stat_scores_multiple_classes(x, y, pos_labels=pos_labels)
+        >>> tps
+        tensor([1.])
+        >>> fps
+        tensor([0.])
+        >>> tns
+        tensor([2.])
+        >>> fns
+        tensor([1.])
+        >>> sups
+        tensor([2.])
+
     """
     if pred.ndim == target.ndim + 1:
+        # Convert onehot pred tensor to categorical
         pred = to_categorical(pred, argmax_dim=argmax_dim)
+    elif pred.ndim == target.ndim == 1:
+        # Convert numerical pred tensor to categorical
+        if torch.is_floating_point(pred):
+            pred = pred.round().long()
```

Comment:
I think I am facing a precision error posed by ```torch.round```

Reply:
maybe you can separate this functionality as an extra function and add some code sample aka doctest to help debug what is happening there